src/handle_subscribe.c

```diff
@@ -112,7 +112,7 @@ int handle__subscribe(struct mosquitto_db *db, struct mosquitto *context)
 			log__printf(NULL, MOSQ_LOG_DEBUG, "\t%s (QoS %d)", sub, qos);
 
 			if(context->protocol == mosq_p_mqtt311){
-				rc = mosquitto_acl_check(db, context, sub, MOSQ_ACL_SUBSCRIBE);
+				rc = mosquitto_acl_check(db, context, sub, 0, NULL, 0, false, MOSQ_ACL_SUBSCRIBE);
```

Comment:
surely you would provide the requested qos here, not 0?

Reply:
Good point.. missed that. :)